engine/src/main/java/org/hibernate/validator/internal/util/DomainNameUtil.java

```diff
@@ -0,0 +1,98 @@
+	private static final String DOMAIN_LABEL = "[a-z0-9!#$%&'*+/=?^_`{|}~-]";
+	private static final String DOMAIN = DOMAIN_LABEL + "+(\\." + DOMAIN_LABEL + "+)*";
+	private static final String IP_DOMAIN = "\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\]";
+	//IP v6 regex taken from http://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses
+	private static final String IP_V6_DOMAIN = "(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))";
+	private static final int MAX_LOCAL_PART_LENGTH = 64;
+
+	/**
+	 * Regular expression for the local part of an email address (everything before '@')
+	 */
+	private static final Pattern LOCAL_PART_PATTERN = Pattern.compile(
+			"(" + LOCAL_PART_ATOM + "+|\"" + LOCAL_PART_INSIDE_QUOTES_ATOM + "+\")" +
+					"(\\." + "(" + LOCAL_PART_ATOM + "+|\"" + LOCAL_PART_INSIDE_QUOTES_ATOM + "+\")" + ")*", CASE_INSENSITIVE
+	);
+
+	/**
+	 * Regular expression for the domain part of an email address (everything after '@')
+	 */
+	private static final Pattern DOMAIN_PATTERN = Pattern.compile(
+			DOMAIN + "|" + IP_DOMAIN + "|" + "\\[IPv6:" + IP_V6_DOMAIN + "\\]", CASE_INSENSITIVE
+	);
+
+	private DomainNameUtil() {
+	}
+
+	/**
+	 * Checks validity of domain name.
+	 *
+	 * @param domain domain to check for validity
+	 *
+	 * @return {@code true} if provided string is a valid domain, {@code false} otherwise
+	 */
+	public static boolean isValidDomainAddress(String domain) {
+		// if we have a trailing dot the domain part we have an invalid email address.
+		// the regular expression match would take care of this, but IDN.toASCII drops the trailing '.'
+		if ( domain.endsWith( "." ) ) {
+			return false;
+		}
+
+		String asciiString;
+		try {
+			asciiString = IDN.toASCII( domain );
+		}
+		catch (IllegalArgumentException e) {
+			return false;
+		}
+
+		if ( asciiString.length() > MAX_DOMAIN_PART_LENGTH ) {
+			return false;
+		}
+
+		Matcher matcher = DOMAIN_PATTERN.matcher( asciiString );
+		return matcher.matches();
+	}
+
+	/**
+	 * Checks validity of emails local part.
+	 *
+	 * @param localPart an email local part to check
+	 *
+	 * @return {@code true} if a given emails local part is valid, {@code false} otherwise
+	 */
+	public static boolean isValidEmailLocalPart(String localPart) {
```

Comment:
I would keep this in the EmailValidator as it's in no way related to the domain validation.